### PR TITLE
Adding Google Analytics API

### PR DESCRIPTION
**Pull request by hgpadovani**
Adding GA API with one query

### CODE REVIEW

File: API Google Analytics/HelloAnalytics.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+from oauth2client import client
+from oauth2client import file
+from oauth2client import tools
+
+SCOPES = ['https://www.googleapis.com/auth/analytics.readonly']
+DISCOVERY_URI = ('https://analyticsreporting.googleapis.com/$discovery/rest')
+KEY_FILE_LOCATION = 'client_secrets.p12'
+SERVICE_ACCOUNT_EMAIL = ''
+VIEW_ID = ''
+
```
Comment (Guilhermeslucas) [inline API Google Analytics/HelloAnalytics.py]: Why use list on line 14, tuple on line 15 and standart string on line 16? Besides, I'd prefer to include this information inside a authentication function. I'm not a big fan of global variables, makes it harder to understand the code

File: API Google Analytics/HelloAnalytics.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+def initialize_analyticsreporting():
+  """Initializes an analyticsreporting service object.
+
+  Returns:
+    analytics an authorized analyticsreporting service object.
+  """
+
```
Comment (Guilhermeslucas) [inline API Google Analytics/HelloAnalytics.py]: Maybe here would be a great point to insert lines between 14 and 18.

File: API Google Analytics/HelloAnalytics.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    }
+    ],
+    "dimensions": [
+    {
+    "name": "ga:campaign"
+    }
+    ]
+    }
+    ]}).execute()
+
```
Comment (Guilhermeslucas) [inline API Google Analytics/HelloAnalytics.py]: This part is confusing me. Can you write this part a little more standardized? Besides, I think it's more straight forward defining the dict and then write the return statement. Besides, as I said earlier, this function can generate some exception too, we'd better raise them.

File: API Google Analytics/HelloAnalytics.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        print( 'Date range (' + str(i) + ')')
+        for metricHeader, value in zip(metricHeaders, values.get('values')):
+          print (metricHeader.get('name') + ': ' + value)
+
+def to_json(response):
+  """ Function that saves response into json file
+  Input: 
+      response
+  """
+  write_json = open('response.json', 'w')
```
Comment (Guilhermeslucas) [inline API Google Analytics/HelloAnalytics.py]: What do you think about letting the one who called this method insert a custom name to the file?

File: API Google Analytics/HelloAnalytics.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+  credentials = ServiceAccountCredentials.from_p12_keyfile(
+    SERVICE_ACCOUNT_EMAIL, KEY_FILE_LOCATION, scopes=SCOPES)
+
+  http = credentials.authorize(httplib2.Http())
+
+  # Build the service object.
+  analytics = build('analytics', 'v4', http=http, discoveryServiceUrl=DISCOVERY_URI)
+
+  return analytics
+
```
Comment (Guilhermeslucas) [inline API Google Analytics/HelloAnalytics.py]: This function looks like a exception generator. What do you think about we use try, except and raise statements.

File: API Google Analytics/HelloAnalytics.py [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+      dateRangeValues = row.get('metrics', [])
+
+      for header, dimension in zip(dimensionHeaders, dimensions):
+        print (header + ': ' + dimension)
+
+      for i, values in enumerate(dateRangeValues):
+        print( 'Date range (' + str(i) + ')')
+        for metricHeader, value in zip(metricHeaders, values.get('values')):
+          print (metricHeader.get('name') + ': ' + value)
+
```
Comment (Guilhermeslucas) [inline API Google Analytics/HelloAnalytics.py:106]: Is there a way to avoid looping so much?

**Review by Guilhermeslucas — CHANGES_REQUESTED** [review]
Really good code, clean, easy to read and understand. Just some minor fixes to be done.

File: API Google Analytics/HelloAnalytics.py (hunk 3 above)
Comment (hgpadovani) [reply API Google Analytics/HelloAnalytics.py]: If I write the dict outise the return statement I got an error:
return analytics.reports().batchGet(body).execute()
TypeError: method() takes 1 positional argument but 2 were given

The only way it works is if I write it inside the return statement, or I'm missing something.

File: API Google Analytics/HelloAnalytics.py (hunk 6 above)
Comment (hgpadovani) [reply API Google Analytics/HelloAnalytics.py:106]: This is the unpacking code proposed by the official documentation of the API, maybe there is a better way to do it, but I don't think we are gonna use this function anyway

File: API Google Analytics/HelloAnalytics.py (hunk 4 above)
Comment (hgpadovani) [reply API Google Analytics/HelloAnalytics.py]: Did you mean passing some filename in the function call?

File: API Google Analytics/HelloAnalytics.py (hunk 6 above)
Comment (Guilhermeslucas) [reply API Google Analytics/HelloAnalytics.py:106]: Understood. So, looks fine for me.

**Pull request merged**